contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,334 @@
+        uint256 amount
+    );
+
+    event ForfeitPaid(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount
+    );
+
+    event WithDraw(
+        address indexed player,
+        uint amount
+    );
+
+    event PlayerOneEndsGame(
+        bytes32 indexed gameKey,
+        address indexed player,
+        uint256 amount
+    );
+
+    // The creator's move is embedded in the game key
+    function movePlayerOne(bytes32 gameKey, uint256 stake) external payable whenNotPaused {
+
+        require(gameKey != bytes32(0), "Game key hash is empty");
+        Game storage game = games[gameKey];
+        require(game.step == Steps.UNSTARTED, INVALID_STEP_MSG);
+        require(game.playerOne == NULL_ADDRESS, INVALID_MOVE_MSG);
+
+        game.stake = stake;
+        game.playerOne = msg.sender;
+        game.step = Steps.PLAYER_ONE_MOVED;
+        depositStake(stake);
+
+        emit PlayerOneMoves(gameKey, msg.sender, stake, msg.value);
+    }
+
+    function movePlayerTwo(bytes32 gameKey, bytes32 moveHash) external payable whenNotPaused {
+
+        require(moveHash != bytes32(0), "Move hash is empty");
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_MOVED, INVALID_STEP_MSG);
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.playerTwo = msg.sender;
+        game.playerTwoMoveHash = moveHash;
+        game.step = Steps.PLAYER_TWO_MOVE;
+        game.expiryDate = expiryDate;
+        depositStake(game.stake);
+
+        emit PlayerTwoMoves(gameKey, msg.sender, msg.value, moveHash, expiryDate);
+    }
+
+    function depositStake(uint256 stake) internal {
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        uint256 senderBalance = balances[msg.sender];
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value).sub(stake);
+    }
+
+    function revealPlayerOne(bytes32 secret, uint8 playerOneMove) external whenNotPaused {
+
+        require(Moves(playerOneMove) >= Moves.ROCK && Moves(playerOneMove) <= Moves.SCISSORS, INVALID_MOVE_MSG);
+        bytes32 gameKey = createPlayerOneMoveHash(msg.sender, secret, Moves(playerOneMove));
+        Game storage game = games[gameKey];
+
+        require(game.step == Steps.PLAYER_TWO_MOVE, INVALID_STEP_MSG);
+        game.step = Steps.PLAYER_ONE_REVEAL;
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.expiryDate = expiryDate;
+
+        game.playerOneMove = Moves(playerOneMove);
+
+        emit PlayerOneReveals(gameKey, msg.sender, Moves(playerOneMove), expiryDate);
+    }
+
+    function revealPlayerTwo(bytes32 gameKey, bytes32 secret, Moves playerTwoMove) external whenNotPaused {
+
+        require(Moves(playerTwoMove) > Moves.EMPTY && Moves(playerTwoMove) <= Moves.SCISSORS, INVALID_MOVE_MSG);
```

Comment:
Now that you have declared it as `, Moves playerTwoMove)`:

1. Why cast it to `Moves` when `playerTwoMove` is already a `Moves`.
2. Have you tried removing the `&& Moves(playerTwoMove) <= Moves.SCISSORS` piece? Maybe it is taken care of by the compiler ;)

Reply:
> 1. Why cast it to `Moves` when `playerTwoMove` is already a `Moves`.

For some reason, in the very early hours, I got in in my head that it would only work with this casting. In the light if day, it works fine without!

Reply:
> 2. Have you tried removing the `&& Moves(playerTwoMove) <= Moves.SCISSORS` piece? Maybe it is taken care of by the compiler ;)

Sorry. Noticed that in `revealPlayerOne` but overlooked here.